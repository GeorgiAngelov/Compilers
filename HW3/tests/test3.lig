var x : int;
var x : int;

// Both of these can be totally ignored for type checking purposes, even if
// that means missing the clash with the next typename.
type x : x;
type x : int;

type x : [int];

function f (x : int) { }
function f (x : bool) : int { }

function x (x : x) : x { }
