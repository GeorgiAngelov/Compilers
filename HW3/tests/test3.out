(decls
   (decl-var:ok (id x) (type-int))
   (decl-var:NONE (id x) (type-int))
   (decl-type:NONE (id x) (type-id x))
   (decl-type:NONE (id x) (type-int))
   (decl-type:ok (id x) (type-array (type-int)))
   (decl-fun:ok (id f) (type-fun (params (typed-id (id x) (type-int))) (type-void)))
   (decl-fun:NONE (id f) (type-fun (params (typed-id (id x) (type-bool))) (type-int)))
   (decl-fun:ok (id x) (type-fun (params (typed-id (id x) (type-id x))) (type-id x))))
Function main() not defined or has wrong type!
Typechecking failed!
